services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,16 +2,48 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import taskcluster from 'taskcluster-client';
 
 export default ({ server, schema, context, path }) =>
   SubscriptionServer.create(
     {
       schema,
       execute,
       subscribe,
-      onConnect(params, socket) {
-        return new Promise(resolve => {
-          credentials()(socket.upgradeReq, {}, resolve);
+      async onConnect(params, socket) {
+
+        return new Promise((resolve, reject) => {
+          credentials()(socket.upgradeReq, {}, async () => {
+            try {
+
+              const token = params.Authorization;
+
+              if (!token) {
+                reject();
+              }
+
+              const credentials= JSON.parse(Buffer.from(token.split(' ')[1], 'base64').toString());
+
+              const authClient = new taskcluster.Auth({
+                rootUrl: process.env['TASKCLUSTER_ROOT_URL'],
+                credentials: credentials
+              });
+
+              const scopes = await authClient.currentScopes();
+              const expandedScopes = await authClient.expandScopes(scopes);
+
+              if (!scopeUtils.satisfiesExpression(scopes.scopes, { AllOf: ['web:read-pulse:*'] })) {
+                await authClient.disableClient(token.clientId);
```

Comment:
That would be too harsh :) If you can't access some parts of application it doesn't mean you should be banned immediately ;)

Reply:
Removed this.